src/test/ly/stealth/mesos/kafka/SchedulerTest.scala

```diff
@@ -234,4 +234,51 @@ class SchedulerTest extends MesosTestCase {
     assertArrayEquals(Array[AnyRef]("1"), Scheduler.otherTasksAttributes("a").asInstanceOf[Array[AnyRef]])
     assertArrayEquals(Array[AnyRef]("2", "3"), Scheduler.otherTasksAttributes("b").asInstanceOf[Array[AnyRef]])
   }
+
+  @Test
+  def onFrameworkMessage = {
+    val broker0 = Scheduler.cluster.addBroker(new Broker("0"))
+    broker0.active = true
+    val broker1 = Scheduler.cluster.addBroker(new Broker("1"))
+    broker1.active = true
+
+    assertNull(broker0.metrics)
+    assertNull(broker1.metrics)
+
+    val metrics0 = new Broker.Metrics()
+    metrics0.underReplicatedPartitions = 2
+    metrics0.offlinePartitionsCount = 3
+    metrics0.activeControllerCount = 1
+    metrics0.timestamp = System.currentTimeMillis()
+
+    val data = scala.util.parsing.json.JSONObject(Map("metrics" -> metrics0.toJson)).toString().getBytes
+
+    Scheduler.frameworkMessage(schedulerDriver, executorId(Broker.nextExecutorId(broker0)), slaveId(), data)
+
+    // metrics updated for corresponding broker
+    assertNotNull(broker0.metrics)
+    assertNull(broker1.metrics)
+
+    def assertMetricsEquals(expected: Broker.Metrics, actual: Broker.Metrics): Unit = {
+      assertEquals(expected.underReplicatedPartitions, actual.underReplicatedPartitions)
+      assertEquals(expected.offlinePartitionsCount, actual.offlinePartitionsCount)
+      assertEquals(expected.activeControllerCount, actual.activeControllerCount)
+      assertEquals(expected.timestamp, actual.timestamp)
+    }
+
+    assertMetricsEquals(metrics0, broker0.metrics)
+
+    // metrics updated only for active brokers
+    broker1.active = false
```

Comment:
this is cool, but the state we need to review that as we have seen a broker be active but it is not true in edge case, i have to go back and see if there is a jira or github for that if not will see about trying to reproduce it again (may have been fixed too but lets use this for a double check). In my view broker == active == true || false is alot of what we are trying to ascertain here. we need some more details in the state so we can ascertain different nuances from "active"=true healthy and "active"=true not healthy, other considerations like that warrant looking at that mechanism again @dmitrypekar not sure this is a huge priority but let me know if you don't get my reasoning behind it please